prov/tcp/src/xnet_progress.c

```diff
@@ -551,7 +557,9 @@ int xnet_start_recv(struct xnet_ep *ep, struct xnet_xfer_entry *rx_entry)
 	assert(xnet_progress_locked(xnet_ep2_progress(ep)));
 	if (!dlist_empty(&ep->unexp_entry)) {
 		dlist_remove_init(&ep->unexp_entry);
-		xnet_update_pollflag(ep, POLLIN, true);
+		ret = xnet_update_pollflag(ep, POLLIN, true);
+		if (ret)
+			goto poll_err;
```

Comment:
This flow looks okay to me (for now).  I think the error propagates back and disables the endpoint.  But, I have a request from middleware for truncation errors to report an error completion without disabling the ep.  That can be handled later, and likely requires more work anyway (like flushing the truncated data from the tcp stream).

Reply:
Ack.